green_calibration.py

```diff
@@ -27,23 +23,58 @@ def __init__(self, csv_data='green_data.csv'):
         self.low_green = np.array([68, 100, 50])
         self.high_green = np.array([84, 255, 255])
 
-    def get_new_hsv(self, mask):
-        if (len(mask) == 0):
-            return self.low_green, self.high_green
+    def get_new_hsv(self, hsv, contours):
+        if len(contours) == 0:
+            return
+
+        # Creates empty mask with shape (480, 640, 1)
+        mask = np.zeros((480, 640, 1), dtype=np.uint8)
+
+        # Populates empty mask with the contour of the target
+        cv2.drawContours(mask, [contours[0]], 0, (255, 255, 255), thickness=cv2.FILLED)
+
+        # List of all HSV values inside the contour
+        greens = hsv[np.where((mask[:, :, 0] == 255))]
+
+        if len(greens) == 0:
+            return
+
         for i in range(100):
-            row = random.randrange(0, len(mask))
-            self.true_green_vals = np.append(self.true_green_vals, np.reshape(np.array(mask[row]), (1, 1, 3)), 0)
+            row = random.randrange(0, len(greens))
+            self.true_green_vals = np.append(self.true_green_vals, np.reshape(np.array(greens[row]), (1, 1, 3)), 0)
 
         h = self.true_green_vals[:, :, 0]
         s = self.true_green_vals[:, :, 1]
         v = self.true_green_vals[:, :, 2]
-        low_h, low_s, low_v = (h.mean() - self.H_STD_TOLERANCE * h.std()), (s.mean() - self.S_STD_TOLERANCE * s.std()), (v.mean() - self.V_STD_TOLERANCE * v.std())
-        high_h, high_s, high_v = (h.mean() + self.H_STD_TOLERANCE * h.std()), (s.mean() + self.S_STD_TOLERANCE * s.std()), (v.mean() + self.V_STD_TOLERANCE * v.std())
+
+        low_h, low_s, low_v = (h.mean() - self.H_STD_TOLERANCE * h.std()),\
```

Comment:
Can we break these out into one per line? Stacking everything in tuples is really hard to read and doesn't get us anything.